src/main/java/org/folio/edge/oaipmh/OaiPmhHandler.java

```diff
@@ -64,11 +94,75 @@ protected void handle(RoutingContext ctx) {
     }
 
     handleCommon(ctx, new String[0], new String[0] , (okapiClient, params) -> {
-      oaiPmhClient = new OaiPmhOkapiClient(okapiClient);
-      performCall(ctx, oaiPmhClient, null);
+      this.okapiClient = okapiClient;
+      getTenants(okapiClient)
+        .thenAccept(list -> {
+          if (isEmpty(list)) {
+            notFound(ctx, "Tenants list is absent or empty");
+          }
+          if (isSingleTenantHarvesting(list)) {
+            oaiPmhClient = new OaiPmhOkapiClient(okapiClient);
+            performCall(ctx, oaiPmhClient, null);
+          } else {
+            if (isFirstRequest(request)) {
+              callToTenant(ctx, list.get(0));
+            } else {
+              var resumptionTokenParams = parseResumptionToken(request.params().get(RESUMPTION_TOKEN));
+              if (shouldStartHarvestingForNextTenant(resumptionTokenParams)) {
+                request.params().remove(RESUMPTION_TOKEN);
+                request.params().set(METADATA_PREFIX, resumptionTokenParams.get(METADATA_PREFIX));
+              }
+              callToTenant(ctx, resumptionTokenParams.get(TENANT_ID));
+            }
+          }
+        });
     });
   }
 
+  private CompletableFuture<List<String>> getTenants(OkapiClient okapiClient) {
+    var res = tenantsCache.get(okapiClient.tenant);
+    if (isEmpty(res)) {
+      return new ConsortiaTenantClient(okapiClient).getConsortiaTenants(null)
+        .toCompletionStage().toCompletableFuture()
+        .thenApply(l -> tenantsCache.put(okapiClient.tenant, l).value);
+    }
+    return CompletableFuture.completedFuture(res);
+  }
+
+  private boolean isSingleTenantHarvesting(List<String> tenants) {
+    return tenants.size() == 1;
+  }
+
+  private boolean shouldStartHarvestingForNextTenant(Map<String, String> params) {
+    return params.size() == 2 && params.containsKey(METADATA_PREFIX) && params.containsKey(TENANT_ID);
+  }
+
+  private void callToTenant(RoutingContext ctx, String tenant) {
+    var client = ocf.getOkapiClient(tenant);
+    getToken(ctx, tenant, client)
+      .thenAccept(token -> {
+        client.setToken(token);
+        oaiPmhClient = new OaiPmhOkapiClient(client);
+        performCall(ctx, oaiPmhClient, null);
+      });
+  }
+
+  private Optional<String> getNextTenant(List<String> list, String tenant) {
+    var nextTenantIndex = list.indexOf(tenant) + 1;
+    if (list.size() > nextTenantIndex) {
+      return Optional.of(list.get(nextTenantIndex));
+    }
+    return Optional.empty();
+  }
+
+  private Map<String, String> parseResumptionToken(String resumptionToken) {
+    var decodedToken = new String(Base64.getUrlDecoder().decode(resumptionToken),
+      StandardCharsets.UTF_8);
+    return URLEncodedUtils
+      .parse(decodedToken, UTF_8, '&').stream()
```

Comment:
PARAMETER_DELIMITER can be used instead of hardcoded '&'?

Reply:
Good catch, fixed